classes/class2/README.md

```diff
@@ -1 +1,303 @@
+[Function: func1]
+1
+```
+
+The two print statements we added to `functions.js` look almost the same, but the console output shows that they're very different. The first statement prints the function object `func1` -- that is, the actual *entity* that represents the function itself (the code, the arguments, the scope in which it was declared -- more on scope later). Node says that `func1` is a `Function` named `func1` -- nothing unexpected here.
+
+Putting `()` next to the name of a function calls the function, so when we print `func1()` we get the return value of `func1`, which is `1`.
+
+`()` will actually always attempt to "call" the expression to its left, treating it like a function. So you can get errors like this:
+
+```node
+> 5()
+TypeError: number is not a function
+```
+
+A variable can also store a function declaration, so we could have done this instead:
+
+```node
+var func1 = function() {
+	return 1
+}
+```
+
+The above example wouldn't have acted *quite* the same, though, which brings us to what makes JavaScript functions so special.
+
+### Functions are Objects
+
+The print statements above hinted at this: in JavaScript, functions are objects. So something like this is possible:
+
+###### functions.js
+```node
+function func1() {
+	var func2 = function() {
+		return 2
+	}
+	return func2
+}
+```
+
+What will happen when we call `func1`?
+
+###### functions.js
+```node
+...
+console.log( func1() )
+```
+
+```bash
+❯❯❯ node functions.js
+[Function]
+```
+
+The function `func1` is a named function that returns the variable `func2`. We set `func2` equal to a function definition without a name -- the function on the right hand side of this assignment is known as an **anonymous function**.
+
+So, `func1` returns an anonymous function which returns `2`. How do we get at the return value of the returned function?
+
+###### functions.js
+```node
+...
+console.log( func1()() )
+```
+
+```bash
+❯❯❯ node functions.js
+2
+```
+
+This is totally legal. It might help to break down the expression we just printed:
+
+```node
+func1()() // let's add parentheses to show the order of evaluation
+( func1() )() // first, func1 is called
+( func2 )() // it returns func2
+( function() { return 2 } )() // func2 holds an anonymous function
+2 // the anonymous function is called and returns 2
+```
+
+## Callbacks
+
+Function objects let us pass around pieces of code that can be run at arbitrary times.
+
+Here's a really common situation: A client application requests data from a URL, and wants to process the data when it arrives. There's no way to know how long it will take the data to arrive, and even if there was, we wouldn't want to just block execution and wait until we got the data. It would be great if we could request the data and go about our business until we received it -- then we could run the code that depends on the data.
+
+That's what callbacks are for. Here's an example with real code:
+
+```node
+jQuery.get(url, success)
```

Comment:
You're right, literally no one uses `jQuery` over `$` — I was just trying to avoid jQuery altogether but if you think a brief introduction is warranted I'll add one.